### DISCUSSION TITLE
Need help with creating custom keys

### DISCUSSION THREAD

**Discussion by antoniovmx** [nickcoutsos/keymap-editor#126] [Q&A] [answered]
Hello everyone! I just found out about your keymap editor and I was wondering if you could help me with creating custom keys for the Kinesis Advantage 360 Pro keyboard. Specifically, I would like to create a key that outputs a parenthesis in its normal state and outputs the number nine when the shift key is used. Knowing how to do this will also help me create other custom hotkeys that I have in mind. Please see attached images of my current layout. Thank you in advance for your help!
![Keymap-Editor-Base](https://github.com/nickcoutsos/keymap-editor/assets/10678107/719b1ade-5823-4808-bc6b-b15b147f5a33)
![Keymap-Editor-Num](https://github.com/nickcoutsos/keymap-editor/assets/10678107/1551a881-d272-4808-b1cc-17c9fffb91a8)
![Keymap-Editor-FN](https://github.com/nickcoutsos/keymap-editor/assets/10678107/f4579bf6-1809-4649-8a15-6b6218a614a7)
![Keymap-Editor-M](https://github.com/nickcoutsos/keymap-editor/assets/10678107/5a804e04-a4bb-466f-b317-0b765128b0f8)

**Comment by nickcoutsos** [comment, marked as answer]
Hi there. Your description sounds like the [`mod-morph` behavior](https://zmk.dev/docs/behaviors/mod-morph).

Mod-morph lets you define two behaviors and let the triggering of one or the other be determined by a modifier key being included in the keypress. ZMK doesn't offer a parameterized version of this right now so you would need to do this on a per-key basis. In code it would look like the following:

```dts
        paren_nine: paren_nine {
            compatible = "zmk,behavior-mod-morph";
            label = "PAREN_NINE";
            bindings =
                <&kp LPAR>,
                <&kp N9>;

            #binding-cells = <0>;
            mods = <(MOD_LSFT|MOD_RSFT)>;
        };
```

The keymap editor will let you create this visually:

**Navigate to the Behaviors tab** and make sure that `Enable beta` is checked (because I never got around to declaring the beta test complete), then click _Add New Behavior_ button.
![Screen Shot 2023-07-25 at 4 29 48 PM](https://github.com/nickcoutsos/keymap-editor/assets/1165066/6527ec19-0b19-4bac-8902-b29e6262141f)

**In the popup dialog select mod morph from the list of behavior types and then click _Create_**
![Screen Shot 2023-07-25 at 4 29 54 PM](https://github.com/nickcoutsos/keymap-editor/assets/1165066/a43d0c18-8120-44ae-bffe-5d3327d51f0f)

**Configure the behavior like so**
![Screen Shot 2023-07-25 at 4 31 11 PM](https://github.com/nickcoutsos/keymap-editor/assets/1165066/9f1b3009-d7d7-4011-a0b3-c4981bd536f3)

Note that for additional behaviors you'll need to have unique names.

**Go back to the layers view and select whichever key you want to update, and change the _Behavior_ value to the one you just created**
![Screen Shot 2023-07-25 at 4 31 32 PM](https://github.com/nickcoutsos/keymap-editor/assets/1165066/749342b2-bae7-4b63-a78f-ea0c915c48e2)

**Comment by antoniovmx** [comment]
Hi Nick,
I just wanted to say a big thank you for your detailed instructions on creating a custom key. Your step-by-step guide, complete with screenshots, made the process easy to follow and I’m happy to report that the custom key is working perfectly. I appreciate the time and effort you put into helping me.
